Automattic/woocommerce-payments
# Only show the Fraud & Risk tour if the user clicked on the "Learn more" button

Fixes #6887 

#### Changes proposed in this Pull Request





This PR adds a check for the custom query parameter we add to the WCPay settings URL when the Fraud and Risk Tools learn more button is clicked. If this query parameter is present and the other two checks pass, then the tour is shown, otherwise it isn't. This is done to prevent a sub-standard user experience in the event a merchant has not dismissed the tour yet, doesn't click 'learn more', and goes directly to the settings page where they are greeted by a grayed out, inactive screen for a second before the smooth scroll takes them to the bottom of the page for the tour. 





#### Testing instructions





- Checkout this PR branch
- Build the client: `npm run build:client`
- All tests should pass
- Delete FRT related options:
  - `npm run wp option delete wcpay_frt_discover_banner_settings`
  - `npm run wp option delete wcpay_fraud_protection_welcome_tour_dismissed`
- Visit the WCPay settings page: `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
- The welcome tour should not load.
- Visit the WCPay overview page: `/wp-admin/admin.php?page=wc-admin&path=/payments/overview`
- Click the learn more button on the FRT discoverability banner to go back to WCPay settings.
- Upon page load, the screen should turn gray and smooth scroll to the bottom of the settings page where the Fraud Protection section is. You should see the welcome tour card.
- Dismiss the tour.
- Go back to the WCPay overview page. Click the learn more button once more. 
- Upon page load the screen should remain active, the page should smooth scroll to the Fraud Protection section but the welcome tour should not re-appear.
- Start Jurassic Tube or Ngrok to access your local site via smartphone.
- Delete the welcome tour option again: `npm run wp option delete wcpay_fraud_protection_welcome_tour_dismissed`
- Login to your site admin via smartphone.
- Repeat the test steps above to confirm that the behavior is the same on a smartphone.

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _[Release testing instructions](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-6.3.0#fraud-and-risk-tools-welcome-tour-display-updates)_
